CTOR-1200 Plugin(apps::proxmox::ve::restapi) - Mode(Host discovery) : Add host IP address (#4581)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/virtualization-proxmox-ve-restapi.md

<Tabs groupId="sync">
<TabItem value="Virt-Proxmox-Ve-Restapi-custom" label="Virt-Proxmox-Ve-Restapi-custom">

| Service Alias | Service Template                             | Service Description  | Discovery  |
|:--------------|:---------------------------------------------|:---------------------|:----------:|
| Node-Usage    | Virt-Proxmox-Ve-Node-Usage-Restapi-custom    | Check node usage     | X          |
| Storage-Usage | Virt-Proxmox-Ve-Storage-Usage-Restapi-custom | Check stockage usage | X          |

> | Vm-Usage      | Virt-Proxmox-Ve-Vm-Usage-Restapi-custom      | Check vms usage      | X          |

Reviewer: ```suggestion
| Vm-Usage      | Virt-Proxmox-Ve-Vm-Usage-Restapi-custom      | Check VM usage      | X          |
```

#### Host discovery

| Rule name       | Description                       |
|:----------------|:----------------------------------|

> | Proxmox VM      | Discover Proxmox Virtual machines |

Reviewer: ```suggestion
| Proxmox VM      | Discover Proxmox virtual machines |
```

Please refer to their [official documentation](https://pve.proxmox.com/wiki/Proxmox_VE_API).

IP address and OS information discovery for virtual machines requires the [Qemu-guest-agent](https://pve.proxmox.com/wiki/Qemu-guest-agent) agent to be installed on them.

## Installing the monitoring connector

### Pack

> 

Reviewer: ```suggestion

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

```

3. Apply the **Virt-Proxmox-Ve-Restapi-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro                  | Description                                                                                                                                         | Default value     | Mandatory   |
|:-----------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | PROXMOXAPIUSERNAME     | Set Proxmox VE Username API user need to have this privileges 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'                        |                   | X           |

Reviewer: ```suggestion
| PROXMOXAPIUSERNAME     | Set Proxmox VE username. The API user needs to have the following privileges: 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'                        |                   | X           |
```

| --proto                                    |   Specify https if needed (default: 'https').                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

> | --api-username                             |   Set Proxmox VE Username API user need to have this privileges 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| --api-username                             |   Set Proxmox VE username. The API user needs to have the following privileges: 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```
